malaria24/ona/admin.py

```diff
@@ -222,8 +232,28 @@ def upload_facility_codes_view(self, request):
         if request.method == 'POST':
             form = FacilityUploadForm(request.POST, request.FILES)
             if form.is_valid():
+                file = form.cleaned_data['upload'].read().decode('utf-8')
+
+                try:
+                    file_content_json = json.loads(file)
+                except json.JSONDecodeError:
+                    string_io_file = io.StringIO(file)
+                    next(string_io_file, None)
```

Comment:
What is this line for?

Reply:
Oh its to stop the CSV dictreader from consuming the header as a row, otherwise we end up with a json that looks like this 

```[
  {
    "FacCode": "FacCode",
    "Facility": "Facility",
    "Province": "Province",
    "District": "District",
    "Sub-District (Locality)": "Sub-District",
    "Phase": "Phase"
  },
  {
    "FacCode": "433044",
    "Facility": "Test-Facility-CSV",
    "Province": "Gauteng",
    "District": "Johannesburg",
    "Sub-District (Locality)": "Soweto",
    "Phase": "1"
  }
]

Reply:
I have since updated this with advice Rudi gave me